### PR TITLE
fix(performance): Handle escaped asterisks in search

### PR DESCRIPTION
**Pull request by Zylphrex**
Previously, any asterisks in the search is treated as a wildcard search. This
change ensures that when an asterisk is escaped with a preceding backslash, it
is correctly seen as an literal and does not result in a wildcard search.

### CODE REVIEW

File: static/app/utils/tokenizeSearch.tsx [hunk 1]
```diff
@@ -406,6 +406,5 @@ function escapeTagValue(value: string) {
   // Need to dig deeper to see where exactly it's wrong.
   //
   // astericks (*) is used for wildcard searches
-  // back slaches (\) is used to escape other characters
-  return typeof value === 'string' ? value.replace(/([\*\\])/g, '\\$1') : value;
+  return typeof value === 'string' ? value.replace(/([\*])/g, '\\$1') : value;
```
Comment (Zylphrex) [inline static/app/utils/tokenizeSearch.tsx:409]: To keep this PR small, removing the escaping of backslashes for now.

File: tests/sentry/api/test_event_search.py [hunk 2]
```diff
@@ -398,3 +398,22 @@ def test_is_query_unsupported(self):
             InvalidSearchQuery, ".*queries are not supported in this search.*"
         ):
             parse_search_query("is:unassigned")
+
+    def test_escaping_asterisk(self):
```
Comment (wmak) [inline tests/sentry/api/test_event_search.py:402]: Could we add an xfail for the edgecase where a string ends with a `\`?

File: src/sentry/api/event_search.py [hunk 3]
```diff
@@ -194,6 +197,29 @@ def translate_wildcard(pat: str) -> str:
     return "^" + res + "$"
 
 
+def translate_escape_sequences(string: str) -> str:
```
Comment (wmak) [inline src/sentry/api/event_search.py:200]: Could we add unit tests for this function directly?

**Review by wmak — APPROVED** [review]
2 minor comments, but lgtm

File: tests/sentry/api/test_event_search.py (hunk 2 above)
Comment (Zylphrex) [reply tests/sentry/api/test_event_search.py:402]: added in 0f0c5ff

File: src/sentry/api/event_search.py (hunk 3 above)
Comment (Zylphrex) [reply src/sentry/api/event_search.py:200]: added in 0f0c5ff

File: tests/sentry/search/events/test_filter.py [hunk 4]
```diff
@@ -840,12 +840,10 @@ def test_negated_wildcard(self):
...
         assert get_filter("release:3.1.\\* user.email:\\*@example.com").conditions == [
-            [["match", ["release", "'(?i)^3\\.1\\.\\*$'"]], "=", 1],
-            [["match", ["user.email", "'(?i)^\\*@example\\.com$'"]], "=", 1],
-        ]
-        assert get_filter("release:\\\\\\*").conditions == [
-            [["match", ["release", "'(?i)^\\\\\\*$'"]], "=", 1]
+            ["release", "=", "3.1.*"],
+            ["user.email", "=", "*@example.com"],
         ]
+        assert get_filter("release:\\\\\\*").conditions == [["release", "=", "\\\\*"]]
```
Comment (Zylphrex) [inline tests/sentry/search/events/test_filter.py:846]: These conditions are equal to what they were before. Previously, it was using a regex to match the literal, and now it's just using a plain equals.

**Pull request merged**